src/fsharp/TastPickle.fs

```diff
@@ -1305,11 +1304,27 @@ let u_ILInstr st =
 // Pickle/unpickle for F# types and module signatures
 //---------------------------------------------------------------------------
 
-let p_Map pk pv = p_wrap Map.toList (p_list (p_tup2 pk pv))
+let p_Map_core pk pv (xs: Map<_, _>) st =
+    for x in xs do
+        pk x.Key st
+        pv x.Value st
+
+let p_Map pk pv x st =
+    p_int (Map.count x) st
+    p_Map_core pk pv x st
+
 let p_qlist pv = p_wrap QueueList.toList (p_list pv)
 let p_namemap p = p_Map p_string p
 
-let u_Map uk uv = u_wrap Map.ofList (u_list (u_tup2 uk uv))
+let u_Map_core uk uv n st =
+    Map.ofSeq (seq {
```

Comment:
I'm not sure this will be any faster.   But it's likely not much slower

Reply:
It's faster. Before, it would build the list which is one iteration, then iterate it again to build the map. With the `seq`, it will just iterate once.